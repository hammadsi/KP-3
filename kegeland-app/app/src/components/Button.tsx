import React from 'react';
import {StyleSheet} from 'react-native';
import {Button as PaperButton} from 'react-native-paper';

export type ButtonProps = React.ComponentProps<typeof PaperButton>;

/**
 * Component for rendering a button.
 * @param props the props
 * @see {@link ButtonProps}
 * @see {@link Button}
 */
const Button: React.FC<ButtonProps> = (props) => {
  return (
    <PaperButton
      {...props}
      labelStyle={
        props.labelStyle
          ? [styles.buttonLabel, props.labelStyle]
          : styles.buttonLabel
      }
      contentStyle={
        props.contentStyle
          ? [styles.buttonContent, props.contentStyle]
          : styles.buttonContent
      }
    />
  );
};

const styles = StyleSheet.create({
  buttonContent: {
    paddingVertical: 5,
  },
  buttonLabel: {
    fontSize: 16,
  },
});

export default Button;
