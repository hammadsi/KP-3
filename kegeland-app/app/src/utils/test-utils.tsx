/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
