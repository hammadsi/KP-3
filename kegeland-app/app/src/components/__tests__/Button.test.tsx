import React from 'react';

import {render} from '~utils/test-utils';

import Button from '../Button';

describe('Test Button-component', () => {
  it('should render correctly', () => {
    const component = <Button>1</Button>;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
