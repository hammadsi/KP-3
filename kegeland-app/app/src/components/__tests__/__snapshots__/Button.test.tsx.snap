// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 15,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 15,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "paddingVertical": 5,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#00b0ff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 16,
                },
              ],
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
</View>
`;
